src/client/Microsoft.Identity.Client/Internal/Requests/ManagedIdentityAuthRequest.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Cache.Items;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Extensibility;
+using Microsoft.Identity.Client.Instance;
+using Microsoft.Identity.Client.ManagedIdentity;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.Internal.Requests
+{
+    internal class ManagedIdentityAuthRequest : RequestBase
+    {
+        private readonly AcquireTokenForManagedIdentityParameters _managedIdentityParameters;
+
+        public ManagedIdentityAuthRequest(
+            IServiceBundle serviceBundle,
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenForManagedIdentityParameters managedIdentityParameters)
+            : base(serviceBundle, authenticationRequestParameters, managedIdentityParameters)
+        {
+            _managedIdentityParameters = managedIdentityParameters;
+        }
+
+        protected override async Task<AuthenticationResult> ExecuteAsync(CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(_managedIdentityParameters.Resource))
+            {
+                throw new MsalClientException(
+                    MsalError.ScopesRequired,
+                    MsalErrorMessage.ScopesRequired);
+            }
+
+            var logger = AuthenticationRequestParameters.RequestContext.Logger;
+            CacheRefreshReason cacheInfoTelemetry = CacheRefreshReason.NotApplicable;
+
+            if (!_managedIdentityParameters.ForceRefresh)
+            {
+                MsalAccessTokenCacheItem cachedAccessTokenItem = await CacheManager.FindAccessTokenAsync().ConfigureAwait(false);
+
+                if (cachedAccessTokenItem != null)
```

Comment:
Will managed identity tokens have a `refreshIn` value?
Should we do a background refresh like what we do for client credential?

https://github.dev/AzureAD/microsoft-authentication-library-for-dotnet/blob/5807f11c53380c6eff90ab6ecc41ae2eb7fbed13/src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs#L98-L108

Reply:
The response for managed identity only have `access token` and `expires on` value. It does not have any refresh token or refresh in value. So that flow is not relevant here.